In [1]:
import pandas as pd

/var/folders/_n/88nsmwlj1k5fmf9psfmkbhfc0000gn/T/ipykernel_12420/4080736814.py:1: DeprecationWarning: 
Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
but was not found to be installed on your system.
If this would cause problems for you,
please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466
        
  import pandas as pd


In [13]:
rawDF = pd.read_csv("dessertSurvey.csv")

rawDF.head()

Timestamp               Email Address What is your name?  \
0  1/24/2024 19:18:47               cvann@uab.edu              chloe   
1  1/24/2024 19:23:39       bbarnard073@gmail.com      Bryce Barnard   
2  1/24/2024 19:25:44        mbyers1023@gmail.com        Maddy Byers   
3  1/24/2024 19:27:07       shraderabby@gmail.com        Abby Grace    
4  1/24/2024 19:28:25  abb.mize@vikings.berry.edu          Abby Mize   

                        What is your major/minor?  How old are you?  \
0                psych major, neuroscience minor                 21   
1                         Biomedical engineering                 22   
2            Early Childhood/Elementary Education                21   
3  Communications / PR (minor: creative writing)                 20   
4                International Affairs/Sociology                 20   

  If you are in college, what year are you? What is your gender?  \
0                                    Junior               Female   
1                                    Junior                 Male   
2                                    Junior               Female   
3                                    Junior               Female   
4                                    Junior               Female   

  What is your favorite dessert? How often/when do you have dessert?  \
0                      Ice Cream                               Daily   
1                      Ice Cream                              Rarely   
2                      Ice Cream                              Weekly   
3                        Cookies                              Weekly   
4                     Cheesecake                              Weekly   

  Do you prefer sweet or savory desserts?  \
0                                   Sweet   
1                                   Sweet   
2                                   Sweet   
3                                   Sweet   
4                                 Depends   

  Do you have any dietary restrictions that influence your dessert choices?  \
0                                                Yes                          
1                                                 No                          
2                                                Yes                          
3                                                 No                          
4                                                 No                          

  Do you prefer homemade or store bought desserts?  \
0                   Depends on the type of dessert   
1                                         Homemade   
2                   Depends on the type of dessert   
3                             No strong preference   
4                                         Homemade   

  Do you find yourself craving dessert?  \
0                                   Yes   
1                             Sometimes   
2                             Sometimes   
3                                   Yes   
4                             Sometimes   

  Where are you more likely to have dessert?  
0                                    at home  
1                                At an event  
2                                At a bakery  
3                                At a bakery  
4                                At an event

PERCENTAGES FROM WHO QUESTIONS

WHO Q1: If you are in college, what year are you?

In [15]:
college_responses = rawDF[rawDF['If you are in college, what year are you?'] != 'Not in college lol']

college_year_counts = college_responses['If you are in college, what year are you?'].value_counts()

college_year_percentages = college_year_counts / college_responses.shape[0] * 100

print("Percentage of responses by college year:")
print(college_year_percentages.apply(lambda x: f"{x:.2f}%"))

Percentage of responses by college year:
If you are in college, what year are you?
Junior                           55.26%
Freshman                         13.16%
Sophomore                        10.53%
Senior                            7.89%
Unenrolled                        2.63%
Senior +                          2.63%
graduated                         2.63%
Drop out                          2.63%
None (Should not be required)     2.63%
Name: count, dtype: object


WHO A1: 


Junior     =                      55.26% 

Freshman           =              13.16% 

Sophomore     =                    10.53% 

Senior             =               7.89% 

Unenrolled          =              2.63% 

Senior +        =                  2.63% 

graduated              =           2.63% 

Drop out              =            2.63% 

None (Should not be required)     2.63%

WHO Q2: How old are you?

In [11]:
age_counts = rawDF['How old are you?'].value_counts()

age_percentages = age_counts / rawDF.shape[0] * 100

print("Percentage of responses by age category:")
print(age_percentages.apply(lambda x: f"{x:.2f}%"))

Percentage of responses by age category:
How old are you?
20    35.90%
21    30.77%
19    15.38%
18     7.69%
22     5.13%
23     5.13%
Name: count, dtype: object


WHO A2: 


20    35.90%

21    30.77%

19    15.38%

18     7.69%

22     5.13%

23     5.13%

WHO Q3: What is your gender?

In [9]:
gender_counts = rawDF['What is your gender?'].value_counts()

gender_percentages = gender_counts / rawDF.shape[0] * 100

print("Percentage of responses by gender category:")
print(gender_percentages.apply(lambda x: f"{x:.2f}%"))

Percentage of responses by gender category:
What is your gender?
Female    64.10%
Male      35.90%
Name: count, dtype: object


WHO A3: 


Female     64.10%

Male      35.90%

PERCENTAGES FROM WHEN QUESTIONS

WHAT Q1: How often/when do you have dessert?

In [16]:
response_counts = rawDF['How often/when do you have dessert?'].value_counts()

response_percentages = response_counts / rawDF.shape[0] * 100

print("Percentage of responses for 'How often/when do you have dessert?':")
print(response_percentages.apply(lambda x: f"{x:.2f}%"))

Percentage of responses for 'How often/when do you have dessert?':
How often/when do you have dessert?
Weekly                  41.03%
Daily                   20.51%
On special occasions    17.95%
Rarely                  15.38%
Monthly                  5.13%
Name: count, dtype: object


WHEN A1:

Weekly                  41.03%

Daily                   20.51%

On special occasions    17.95%

Rarely                  15.38%

Monthly                  5.13%

WHEN Q2: Where are you more likely to have dessert?

In [17]:
response_counts = rawDF['Where are you more likely to have dessert?'].value_counts()

response_percentages = response_counts / rawDF.shape[0] * 100

print("Percentage of responses for 'Where are you more likely to have dessert?':")
print(response_percentages.apply(lambda x: f"{x:.2f}%"))

Percentage of responses for 'Where are you more likely to have dessert?':
Where are you more likely to have dessert?
At a bakery                              46.15%
At an event                              20.51%
At a restaurant                          12.82%
At home                                   5.13%
at home                                   2.56%
wherever and whenever it is available     2.56%
Anywhere that I can                       2.56%
At home at 12am truthfully                2.56%
anytime i need a sweet treat              2.56%
Home                                      2.56%
Name: count, dtype: object


WHEN A2:

At a bakery                              46.15%

At an event                              20.51%

At a restaurant                          12.82%

At home                                   5.13%

at home                                   2.56%

wherever and whenever it is available     2.56%

Anywhere that I can                       2.56%

At home at 12am truthfully                2.56%

anytime i need a sweet treat              2.56%

Home                                      2.56%

PERCENTAGES OF WHAT QUESTIONS

WHAT Q1:  What is your favorite dessert?


In [18]:
response_counts = rawDF['What is your favorite dessert?'].value_counts()

response_percentages = response_counts / rawDF.shape[0] * 100

print("Percentage of responses for 'What is your favorite dessert?':")
print(response_percentages.apply(lambda x: f"{x:.2f}%"))

Percentage of responses for 'What is your favorite dessert?':
What is your favorite dessert?
Ice Cream        25.64%
Brownies         25.64%
Cookies          17.95%
Cake             10.26%
Pie               5.13%
Cheesecake        2.56%
Cheesecake        2.56%
 Cookie Dough     2.56%
Frozen Yogurt     2.56%
Custards          2.56%
Donuts            2.56%
Name: count, dtype: object


WHAT A1: 

Ice Cream        25.64%

Brownies         25.64%

Cookies          17.95%

Cake             10.26%

Pie               5.13%

Cheesecake        2.56%

Cheesecake        2.56%

 Cookie Dough     2.56%

Frozen Yogurt     2.56%

Custards          2.56%

Donuts            2.56%

WHAT Q2:  Do you prefer sweet or savory desserts?


In [19]:
response_counts = rawDF['Do you prefer sweet or savory desserts?'].value_counts()

response_percentages = response_counts / rawDF.shape[0] * 100

print("Percentage of responses for 'Do you prefer sweet or savory desserts?':")
print(response_percentages.apply(lambda x: f"{x:.2f}%"))

Percentage of responses for 'Do you prefer sweet or savory desserts?':
Do you prefer sweet or savory desserts?
Sweet                                           64.10%
Depends                                         23.08%
Savory                                          10.26%
both!! the perfect combo has both elements:)     2.56%
Name: count, dtype: object


WHAT A2:

Sweet                                           64.10%

Depends                                         23.08%

Savory                                          10.26%

both!! the perfect combo has both elements:)     2.56%


WHAT Q3: Do you have any dietary restrictions that influence your dessert choices?

In [12]:
dietary_counts = rawDF['Do you have any dietary restrictions that influence your dessert choices?'].value_counts()

dietary_percentages = dietary_counts / rawDF.shape[0] * 100

print("Percentage of responses by dietary restriction category:")
print(dietary_percentages.apply(lambda x: f"{x:.2f}%"))

Percentage of responses by dietary restriction category:
Do you have any dietary restrictions that influence your dessert choices?
No     84.62%
Yes    15.38%
Name: count, dtype: object


WHAT A3:

No     84.62%

Yes    15.38%

WHAT Q4:   Do you prefer homemade or store-bought desserts?


In [26]:
response_counts = rawDF['Do you prefer homemade or store bought desserts?'].value_counts()

response_percentages = response_counts / rawDF.shape[0] * 100

print("Percentage of responses for 'Do you prefer homemade or store bought desserts?':")
print(response_percentages.apply(lambda x: f"{x:.2f}%"))

Percentage of responses for 'Do you prefer homemade or store bought desserts?':
Do you prefer homemade or store bought desserts?
Depends on the type of dessert                    38.46%
Homemade                                          38.46%
No strong preference                              15.38%
Store bought                                       5.13%
Homemade unless from Mac’s bakery in Rockmart      2.56%
Name: count, dtype: object


WHAT A4:


Depends on the type of dessert                    38.46%

Homemade                                          38.46%

No strong preference                              15.38%

Store bought                                       5.13%

Homemade unless from Mac’s bakery in Rockmart      2.56%

WHY Q1: Do you find yourself craving dessert?

In [27]:
response_counts = rawDF['Do you find yourself craving dessert?'].value_counts()

response_percentages = response_counts / rawDF.shape[0] * 100

print("Percentage of responses for 'Do you find yourself craving dessert?':")
print(response_percentages.apply(lambda x: f"{x:.2f}%"))

Percentage of responses for 'Do you find yourself craving dessert?':
Do you find yourself craving dessert?
Yes          53.85%
Sometimes    41.03%
No            5.13%
Name: count, dtype: object


WHY A1:

Yes          53.85%

Sometimes    41.03%

No            5.13%

EXTRA DATA ANALYSIS 